### ISSUE TITLE
Better logging of failed fix message parse due to malformed date time

### ISSUE THREAD

**Issue by benbellick** [imandra-ai/fix-engine#216]
For example, imagine the following message is sent: `8=FIX.4.2|9=72|35=A|49=IMANDRA|56=COMP|52=2--0-24-1-009-15:21:38.000|34=50|98=0|108=30|10=051|`

The date field is incorrectly formatted. The fix-engine should log that the date is specifically the cause of the failure, rather than silently failing.

**Comment by sam-tombury** [comment]
Idea being that [these 'transitions'](https://github.com/imandra-ai/fix-engine/blob/9848cbe1ba0fd6e276721604bf015229f485ca2d/fix-engine/src/fix_engine.iml#L215-L229) should emit a transition log which can be observed by the caller

**Comment by benbellick** [comment]
The problem can be resolved roughly by editing the following function in `fix_engine.iml`

```ocaml
(** Process incoming FIX message here. *)
let proc_incoming_fix_msg ( m, engine : full_top_level_msg * fix_engine_state) = 
    match m with
    | Garbled                   -> engine   (* Garbled messages are simply ignored. Note the timestamp is not updated. *)
    | SessionRejectedMsg data   -> 
        begin
            match engine.fe_curr_mode with
            | ActiveSession         -> let state' = session_reject ( data, engine ) in 
                                       { state' with fe_last_data_received = engine.fe_curr_time }
            | _                     -> engine
        end
    | BusinessRejectedMsg data  ->
        begin
            match engine.fe_curr_mode with 
            | ActiveSession         -> let state' = business_reject ( data, engine ) in 
                                       { state' with fe_last_data_received = engine.fe_curr_time }
            | _                     -> engine
                                      |> with_transition_message (MessageIgnored "Session rejected") (*crucial addition here*)
```
